modules/appmodel/tests/test_app_parser_generator.c

```diff
@@ -77,11 +77,48 @@ _build_cfg_args(const gchar *key, const gchar *value, ...)
   return args;
 }
 
+
+static GString *
+_remove_comments(const GString *str)
+{
+  GString *prune = g_string_new("");
+  gboolean is_inside_comment = FALSE;
+
+  for (gint i=0; i < str->len; ++i)
+    {
+      const gchar curr = str->str[i];
+      const gboolean is_comment_start = '#' == curr;
+      const gboolean is_comment_end = curr == '\n';
+      const gboolean does_comment_continue_next_line = is_comment_end && (i+1) < str->len && '#' == str->str[i+1];
```

Comment:
Why do we need this variable? To join the before-after lines of a comment block?

Reply:
The aim is to remove the `\n` from *multi line* comment blocks like this:
```
#Comment1\n
#Comment2\n
log {};
```

Without this it would became:
```
\n
\n
log {};
```

I did not want to change the assert, but rather remove what not needed.
This is actually not perfect, because it would detect as a comment in strings: `file('/tmp/#bla');'

But I would rather avoid parsing, and would require from UT writer to keep that in mind.